cranelift/codegen/src/isa/x64/inst.isle

```diff
@@ -3066,10 +3109,11 @@
 
 ;; Helper for creating `minps` instructions.
 (decl x64_minps (Xmm Xmm) Xmm)
```

Comment:
I don't think we can update this to `XmmMem` since `minps` requires alignment, but its unfortunate since it also prevents loads sinking into the AVX version.

Is there another way around it? I suspect we are going to run into more cases like this in the future.

My thoughts would be to do something like:
```lisp
(decl load_me_maybe (XmmMem) Xmm)

(decl x64_minps (Xmm XmmMem) Xmm)
(rule 0 (x64_minps x y)
      (xmm_rm_r (SseOpcode.Minps) x (load_me_maybe y)))
```

But I'm not sure how good an idea that is, and we definitely don't need to do it in this PR.

Reply:
One idea I had was that in the `sinkable_load` extractor a helper function `is_mergeable_load` is called which currently says `false` for all SIMD-based loads if they aren't flagged as aligned. I was thinking that if we have complete AVX support we might be able to make that function conditional where if AVX is enabled then it allows unaligned loads but if not it disallows unaligned loads. That's still somewhat brittle, though, so I don't honestly know the best way to handle this at this time.

Reply:
Similar to the above comments, we should probably do load sinking in a separate PR since it is a bit more advanced.

Reply:
> That's still somewhat brittle, though, so I don't honestly know the best way to handle this at this time.

We could pass the opcode to `is_mergeable_load` as well and have an allow list for AVX instructions that don't barf on unaligned loads.

Reply:
I like that idea! We've already got a separate path for sinking loads for sse operations (`put_in_xmm_mem`) which would be a great place to add this information. The only downside I can see with changing the signature of that function is that we'd no longer be able to use it as an implicit conversion. I think that might actually be a great change though, as it would force us to be a bit more thoughtful about where we're allowing loads to be sunk for sse instructions.

Reply:
Alternate idea, for future consideration: an instruction that takes a R/M arg but requires alignment could encode that in the type, via e.g. an `XmmMemAligned` arg. (If we currently have a case where we use the same `MInst` enum variant for an alignment-required opcode and not, we should split into separate variants.) Then if we made a pass over all instructions to ensure we got the types right, we can be much more permissive with the automatic conversions everywhere, without having to worry about continued vigilance with explicit `put_in_xmm_mem`, etc.